text/0074/0074-entity-slicing.md

```diff
@@ -0,0 +1,566 @@
+
+```
+pub struct AccessTrie {
+    /// Child data of this entity slice.
+    pub children: HashMap<SmolStr, Box<AccessTrie>>,
+    /// If the ancestors in the entity hierarchy are required.
+    pub ancestors_required: bool
+}
+```
+
+
+
+
+## Computing the Entity Manifest: High Level
+
+Computing the entity manifest has two parts: **calculating access paths** and **merging** **access paths.**
+For calculating access paths, we write a static analysis that finds all the relevant access paths in the policies. Cedar has only four variables (`principal`, `resource`, `action`, and  `context` ), so all data is loaded starting with a variable or entity literal.
+For merging access paths, we define a method for merging access paths, sharing common prefixes in a Trie.
+
+The high-level algorithm looks like this:
+
+1. First, perform Cedar type-checking to get type-annotated policies that are specific to the type of the request. 
+2. For each type of request
+    1. For each policy with that request type
+        1. **Compute access paths**
+    2. **Merge** all of the access paths into a single entity slice trie
+
+
+The merge operator for access paths is straightforward. First, convert all access paths directly to a Trie, where the fields in the path are the edges in the Trie. Then, use the standard merge operator on Tries.
+
+
+## Computing access paths
+
+This section proposes a simple static analysis that soundly computes all of the necessary access paths for a Cedar expression.
+While it’s possible to soundly handle all Cedar expressions, we simplify the problem by rejecting Cedar programs unless they follow the following grammar:
+
+
+```
+<expr> = <datapath-expr>
+         <datapath-expr> in <expr>
+         <expr> + <expr>
+         if <expr> { <expr> } { <expr> }
+         ... all other Cedar operators not mentioned by datapath-expr
+         
+<datapath-expr> = <datapath-expr>.<field>
+                  <datapath-expr> has <field>
+                  <variable>
+                  <entity literal>
+                  
+<variable> = principal 
+             resource
+             action
+             context
+                  
+```
+
+
+The grammar partitions Cedar expressions into **datapath expressions** and all other Cedar operators.
+On the LHS of `in`, `.`, and `has`, expressions involving `in`, `+`, `if-then-else`, and most other Cedar functions/operators are not allowed; only Cedar variables, entity literals, and `.` or `has` expressions are allowed.
+
+The partition is reasonable since, in practice, users do not interleave datapath expressions with other operators.
+Here are examples of Cedar expressions rejected by this grammar:
+
+
+```
+{ "myfield": principal.name }.myfield
+(if principal.ismanager then { "myfield" : 2 } else { "myfield": 3 }).myfield
+(if principal has mobilePhone then principal.mobilePhone else principal.workPhone).zipCode
```

Comment:
nit: I think you mean `areaCode` here 

That being said, the first two expressions in your list seem somewhat contrived, but this one actually seems kind of reasonable at first blush. What makes you say that this expression isn't one that policy authors use in practice?